### PR TITLE
feat(metadata-sidebar): handle delete metadata instance

### PR DESCRIPTION
**Pull request by wpiesiak**
Add delete metadata instance handler to Metadata Sidebar Redesigned

### CODE REVIEW

File: src/common/types/metadata.js [hunk 1]
```diff
@@ -132,6 +132,7 @@ type MetadataTemplateInstance = {
     id: string,
     fields: MetadataTemplateInstanceField[],
     scope: string,
+    templateId?: string,
```
Comment (karolinaru) [inline src/common/types/metadata.js:135]: Do we need to change type of `MetadataTemplateInstance` in `metadata-editor` as well? Or change in `metadata-editor` and import in `Metadata.js`? In other component we import types directly from `metadata-editor` but if I remember correctly this method `createTemplateInstance` was created before we could import

Comment (wpiesiak) [reply src/common/types/metadata.js]: You are right, it seem to be to much hustle with additional attribute. Although, I think MetadataTemplateInstance id field should reflect id of an instance, no a template, so I'll keep this change

File: src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx [hunk 2]
```diff
@@ -106,4 +133,46 @@ describe('useSidebarMetadataFetcher', () => {
...
+    test('should handle metadata instance removal', async () => {
+        mockAPI.getMetadata.mockImplementation((file, successCallback) => {
+            successCallback({ templateInstances: mockTemplateInstances, templates: mockTemplates });
+        });
+        mockAPI.deleteMetadata.mockImplementation((file, template, successCallback) => {
+            successCallback(mockTemplateInstances[0]);
+        });
+
+        const { result } = setupHook();
+        await waitFor(() => expect(result.current.templateInstances).toEqual(mockTemplateInstances));
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx:146]: we dont need the `await waitFor` on any of the `expects`. on the functions like when you do this `await waitFor(() => result.current.deleteMetadataInstance(mockTemplateInstances[0]));` is still required in these cases.

File: src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx [hunk 3]
```diff
@@ -106,4 +133,46 @@ describe('useSidebarMetadataFetcher', () => {
...
+    test('should handle metadata instance removal error', async () => {
+        mockAPI.getMetadata.mockImplementation((file, successCallback) => {
+            successCallback({ templateInstances: mockTemplateInstances, templates: mockTemplates });
+        });
+        mockAPI.deleteMetadata.mockImplementation((file, template, successCallback, errorCallback) => {
+            errorCallback(mockError, 'metadata_remove_error');
+        });
+
+        const { result } = setupHook();
+        await waitFor(() => expect(result.current.status).toBe(STATUS.SUCCESS));
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx:164]:
```suggestion
        expect(result.current.status).toEqual(STATUS.SUCCESS);
```

File: src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx [hunk 4]
```diff
@@ -106,4 +133,46 @@ describe('useSidebarMetadataFetcher', () => {
...
+
+        await waitFor(() => result.current.deleteMetadataInstance(mockTemplateInstances[0]));
+
+        await waitFor(() => expect(result.current.status).toBe(STATUS.ERROR));
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx:168]:
```suggestion
       expect(result.current.status).toEqual(STATUS.ERROR);
```

**Review by jankowiakdawid — APPROVED** [review]

**Review by greg-in-a-box — APPROVED** [review]

**Review by JChan106 — APPROVED** [review]

**Pull request merged**